pkg/splunk/enterprise/clustermaster.go

```diff
@@ -127,6 +128,12 @@ func ApplyClusterMaster(client splcommon.ControllerClient, cr *enterprisev1.Clus
 
 	// no need to requeue if everything is ready
 	if cr.Status.Phase == splcommon.PhaseReady {
+		//upgrade fron automated MC to MC CRD
+		namespacedName := types.NamespacedName{Namespace: cr.GetNamespace(), Name: GetSplunkStatefulsetName(SplunkMonitoringConsole, cr.GetNamespace())}
```

Comment:
Can the namespace be extracted within `splctrl.VerfiyMCStatefulSetOwnerRef` itself? Since. are anyway passing the cr as an argument..

Same comment for other cases too

Reply:
Discussed offline with @kashok-splunk . `GetSplunkStatefulsetName` can't be called in the controller package (where `DeleteReferencesToAutomatedMCIfExists` resides) due to a cyclic dependency, hence the namespace calculation needs to happen outside.